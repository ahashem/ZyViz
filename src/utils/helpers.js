/**
 * Provide shortened version (3 letters) of the provided Day of Week
 * @param weekDay
 * @return {string} shortened weekday
 */
const shortenWeekDay = (weekDay) => {
  const weekDays = {
    'Mondays': 'Mon',
    'Monday': 'Mon',
    'Tuesdays': 'Tue',
    'Tuesday': 'Tue',
    'Wednesdays': 'Wed',
    'Wednesday': 'Wed',
    'Thursdays': 'Thu',
    'Thursday': 'Thu',
    'Fridays': 'Fri',
    'Friday': 'Fri',
    'Saturdays': 'Sat',
    'Saturday': 'Sat',
    'Sundays': 'Sun',
    'Sunday': 'Sun',
  };

  return weekDays[`${weekDay}`];
};

/**
 * Format currency with USD
 * @param amount
 * @return {string}
 */
const currencyTickFormat = (amount) => (((amount / 1000) >= 1) ? `$${(amount / 1000)}K` : `$${amount}`);

export { shortenWeekDay, currencyTickFormat };
