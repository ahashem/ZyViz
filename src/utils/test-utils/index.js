import crossfilter from '../../components/crossfilter/crossfilter';
import NormalizeData from '../../views/CrossVisualizer/visualizer-data-structure/data-normalizer';
import mockOrdersJSON from './mock-test-orders';
import GenerateDimensions from '../../views/CrossVisualizer/visualizer-data-structure/dimensions-manager';

const mockOrdersCrossfilter = crossfilter(NormalizeData(mockOrdersJSON));
const mockOrdersDimensions = GenerateDimensions(mockOrdersCrossfilter);

export { mockOrdersJSON as jsonMock, mockOrdersCrossfilter as crossFilterMock, mockOrdersDimensions as dimensionsMock };
