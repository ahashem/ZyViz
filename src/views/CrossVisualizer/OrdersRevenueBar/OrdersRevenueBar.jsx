import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FlexGridRow from '../../../components/flex-wrapper/FlexGridRow';
import BarChart from '../../../components/BarChart/BarChart';
import { Card, Col } from 'antd';
import { currencyTickFormat, shortenWeekDay } from '../../../utils/helpers';

/**
 * @component OrdersRevenueBar
 */
class OrdersRevenueBar extends Component {
  static propTypes = {
    orders: PropTypes.shape({}),
    dimension: PropTypes.shape({}),
  };

  static defaultProps = {
    orders: {},
    dimension: {},
  };

  /**
   * Receive a crossfilter dimensions, create groups to use for charts
   * @param {Object} dimensions
   * @return {Object<Array<crossfilter.Grouping>>}
   */
  prepareOrdersCountData = (dimensions) => {
    // Serialize and group data using crossfilter to be
    // for-example: {key: 'paymentMethod', value: 20}

    // Orders by Branch
    const ordersByBranch = dimensions.ordersByBranch;
    const ordersByBranchData = ordersByBranch.group().reduceSum(order => order.orderAmount).all();

    // Orders by DeliveryArea
    const ordersByDeliveryArea = dimensions.ordersByDeliveryArea;
    const ordersByDeliveryAreaData = ordersByDeliveryArea.group().reduceSum(order => order.orderAmount).top(20);

    // Orders by Order Day of Week
    // Week days (Saturday - Sunday ...etc)
    const ordersByWeekDay = dimensions.ordersByWeekDay;
    const ordersByWeekDayData = ordersByWeekDay.group().reduceSum(order => order.orderAmount).all();

    return {
      ordersByBranchData,
      ordersByDeliveryAreaData,
      ordersByWeekDayData,
    };

  };

  /**
   * Uplift filter triggers
   * @param {*|string} dimension
   * @param {Number} selectedKey
   * @param {Boolean} deselected
   */
  crossFilterSelected = (dimension, selectedKey, deselected = false) => {
    const { onFilter } = this.props;
    onFilter(dimension, selectedKey, deselected);
  };


  render() {
    const { dimensions, loading } = this.props;
    const ordersCrossed = dimensions ? this.prepareOrdersCountData(dimensions) : null;

    return (
      <div>
        <FlexGridRow justify="space-around" gutter={8}>
          <Col
            xs={24}
            sm={24}
            md={8}
            lg={8}
            xl={8}
            xxl={8}
          >
            <Card title="Revenue By Branch and Week Day" bordered hoverable loading={loading}>
              <BarChart
                height={250}
                width={350}
                labels={{ x: 'Branch', y: 'Revenue' }}
                axisFormats={{
                  x: branch => (branch.toString().replace(/Branch /, '')),
                  y: currencyTickFormat
                }}
                data={ordersCrossed && ordersCrossed.ordersByBranchData}
                onClick={(selectedIndex, deselected) => this.crossFilterSelected(
                  'ordersByBranch',
                  ordersCrossed.ordersByBranchData[selectedIndex].key,
                  deselected
                )}
                x="key"
                y="value"
                labelKey="value"
              />
              <BarChart
                height={250}
                width={350}
                labels={{ x: 'Day of Week', y: 'Revenue' }}
                axisFormats={{ x: shortenWeekDay, y: currencyTickFormat }}
                data={ordersCrossed && ordersCrossed.ordersByWeekDayData}
                onClick={(selectedIndex, deselected) => this.crossFilterSelected(
                  'ordersByWeekDay',
                  ordersCrossed.ordersByWeekDayData[selectedIndex].key,
                  deselected
                )}
                x="key"
                y="value"
                labelKey="value"
              />
            </Card>
          </Col>

          <Card title="Top 20 Delivery Areas By Revenue" bordered hoverable loading={loading}>
            <BarChart
              horizontal
              labeledBars
              hiddenTicks={{ y: true }}
              height={500}
              width={720}
              labels={{ y: 'Delivery Area', x: 'Revenue' }}
              axisFormats={{ x: currencyTickFormat }}
              data={ordersCrossed && ordersCrossed.ordersByDeliveryAreaData}
              onClick={(selectedIndex, deselected) => this.crossFilterSelected(
                'ordersByDeliveryArea',
                ordersCrossed.ordersByDeliveryAreaData[selectedIndex].key,
                deselected
              )}
              x="key"
              y="value"
              labelKey="key"
            />
          </Card>

        </FlexGridRow>
      </div>
    );
  }
}

export default OrdersRevenueBar;
