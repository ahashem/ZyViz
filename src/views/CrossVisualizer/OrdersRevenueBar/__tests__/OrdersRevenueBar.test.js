import React from 'react';
import { shallow } from 'enzyme';

import OrdersRevenueBar from '../OrdersRevenueBar';
import { crossFilterMock, dimensionsMock } from '../../../../utils/test-utils';

describe('<OrdersRevenueBar /> Component', () => {

  test('should render children and match snapshot', () => {
    const orders = crossFilterMock;
    const dimensions = dimensionsMock;
    const onFilter = jest.fn();

    const rendered = shallow(<OrdersRevenueBar
      orders={orders}
      dimensions={dimensions}
      onFilter={onFilter}
    />);

    expect(rendered).toMatchSnapshot();
  });

});
