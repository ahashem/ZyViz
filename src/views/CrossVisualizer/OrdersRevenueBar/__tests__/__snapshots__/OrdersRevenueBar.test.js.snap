// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersRevenueBar /> Component should render children and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrdersRevenueBar
    dimension={Object {}}
    dimensions={
      Object {
        "ordersByBranch": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDate": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByDeliveryArea": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByPaymentMethod": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersBySize": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByTimeOfDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
        "ordersByWeekDay": Object {
          "accessor": [Function],
          "bottom": [Function],
          "currentFilter": [Function],
          "dispose": [Function],
          "filter": [Function],
          "filterAll": [Function],
          "filterExact": [Function],
          "filterFunction": [Function],
          "filterRange": [Function],
          "group": [Function],
          "groupAll": [Function],
          "hasCurrentFilter": [Function],
          "id": [Function],
          "remove": [Function],
          "top": [Function],
        },
      }
    }
    onFilter={[MockFunction]}
    orders={
      Object {
        "add": [Function],
        "all": [Function],
        "allFiltered": [Function],
        "dimension": [Function],
        "groupAll": [Function],
        "isElementFiltered": [Function],
        "onChange": [Function],
        "remove": [Function],
        "size": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FlexGridRow
        gutter={48}
        justify="space-around"
      >
        <BarChart
          axisFormats={
            Object {
              "x": [Function],
              "y": [Function],
            }
          }
          data={
            Array [
              Object {
                "key": "Branch A",
                "value": 230.59,
              },
              Object {
                "key": "Branch B",
                "value": 123.42000000000002,
              },
              Object {
                "key": "Branch C",
                "value": 250.42000000000002,
              },
              Object {
                "key": "Branch D",
                "value": 230.42000000000002,
              },
              Object {
                "key": "Branch E",
                "value": 220.12,
              },
              Object {
                "key": "Branch F",
                "value": 375.95,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
        <BarChart
          axisFormats={
            Object {
              "x": [Function],
              "y": [Function],
            }
          }
          data={
            Array [
              Object {
                "key": "Rai",
                "value": 176.21,
              },
              Object {
                "key": "Qibla",
                "value": 130.1,
              },
              Object {
                "key": "Zahra",
                "value": 112.94,
              },
              Object {
                "key": "Al Qusour",
                "value": 91.43,
              },
              Object {
                "key": "Al Bidae",
                "value": 89.96,
              },
              Object {
                "key": "Hadiya",
                "value": 85.38,
              },
              Object {
                "key": "Abdullah Al Salem",
                "value": 83.43,
              },
              Object {
                "key": "Faiha",
                "value": 83.02,
              },
              Object {
                "key": "Jaber Al Ahmad",
                "value": 78.24,
              },
              Object {
                "key": "Nuzha",
                "value": 74.36,
              },
              Object {
                "key": "Abu Halifa",
                "value": 67.52,
              },
              Object {
                "key": "Sheikh Saad Aviation Terminal",
                "value": 57.89,
              },
              Object {
                "key": "Reggai",
                "value": 47.269999999999996,
              },
              Object {
                "key": "Qortuba",
                "value": 45.3,
              },
              Object {
                "key": "Saad Al Abdullah",
                "value": 36.51,
              },
              Object {
                "key": "Nahda",
                "value": 36.35,
              },
              Object {
                "key": "Mirqab",
                "value": 34.87,
              },
              Object {
                "key": "Fahaheel",
                "value": 34.84,
              },
              Object {
                "key": "Qasser",
                "value": 30.51,
              },
              Object {
                "key": "Doha",
                "value": 20.85,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
        <BarChart
          axisFormats={
            Object {
              "x": [Function],
              "y": [Function],
            }
          }
          data={
            Array [
              Object {
                "key": "Fridays",
                "value": 83.29,
              },
              Object {
                "key": "Mondays",
                "value": 57.89,
              },
              Object {
                "key": "Saturdays",
                "value": 234.60000000000002,
              },
              Object {
                "key": "Sundays",
                "value": 187.74,
              },
              Object {
                "key": "Thursdays",
                "value": 418.09000000000003,
              },
              Object {
                "key": "Tuesdays",
                "value": 65.38,
              },
              Object {
                "key": "Wednesdays",
                "value": 383.93,
              },
            ]
          }
          onClick={[Function]}
          x="key"
          y="value"
        />
      </FlexGridRow>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Branch A",
                  "value": 230.59,
                },
                Object {
                  "key": "Branch B",
                  "value": 123.42000000000002,
                },
                Object {
                  "key": "Branch C",
                  "value": 250.42000000000002,
                },
                Object {
                  "key": "Branch D",
                  "value": 230.42000000000002,
                },
                Object {
                  "key": "Branch E",
                  "value": 220.12,
                },
                Object {
                  "key": "Branch F",
                  "value": 375.95,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Rai",
                  "value": 176.21,
                },
                Object {
                  "key": "Qibla",
                  "value": 130.1,
                },
                Object {
                  "key": "Zahra",
                  "value": 112.94,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 91.43,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 89.96,
                },
                Object {
                  "key": "Hadiya",
                  "value": 85.38,
                },
                Object {
                  "key": "Abdullah Al Salem",
                  "value": 83.43,
                },
                Object {
                  "key": "Faiha",
                  "value": 83.02,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 78.24,
                },
                Object {
                  "key": "Nuzha",
                  "value": 74.36,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 67.52,
                },
                Object {
                  "key": "Sheikh Saad Aviation Terminal",
                  "value": 57.89,
                },
                Object {
                  "key": "Reggai",
                  "value": 47.269999999999996,
                },
                Object {
                  "key": "Qortuba",
                  "value": 45.3,
                },
                Object {
                  "key": "Saad Al Abdullah",
                  "value": 36.51,
                },
                Object {
                  "key": "Nahda",
                  "value": 36.35,
                },
                Object {
                  "key": "Mirqab",
                  "value": 34.87,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 34.84,
                },
                Object {
                  "key": "Qasser",
                  "value": 30.51,
                },
                Object {
                  "key": "Doha",
                  "value": 20.85,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Fridays",
                  "value": 83.29,
                },
                Object {
                  "key": "Mondays",
                  "value": 57.89,
                },
                Object {
                  "key": "Saturdays",
                  "value": 234.60000000000002,
                },
                Object {
                  "key": "Sundays",
                  "value": 187.74,
                },
                Object {
                  "key": "Thursdays",
                  "value": 418.09000000000003,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 65.38,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 383.93,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />,
        ],
        "gutter": 48,
        "justify": "space-around",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "axisFormats": Object {
              "x": [Function],
              "y": [Function],
            },
            "data": Array [
              Object {
                "key": "Branch A",
                "value": 230.59,
              },
              Object {
                "key": "Branch B",
                "value": 123.42000000000002,
              },
              Object {
                "key": "Branch C",
                "value": 250.42000000000002,
              },
              Object {
                "key": "Branch D",
                "value": 230.42000000000002,
              },
              Object {
                "key": "Branch E",
                "value": 220.12,
              },
              Object {
                "key": "Branch F",
                "value": 375.95,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "axisFormats": Object {
              "x": [Function],
              "y": [Function],
            },
            "data": Array [
              Object {
                "key": "Rai",
                "value": 176.21,
              },
              Object {
                "key": "Qibla",
                "value": 130.1,
              },
              Object {
                "key": "Zahra",
                "value": 112.94,
              },
              Object {
                "key": "Al Qusour",
                "value": 91.43,
              },
              Object {
                "key": "Al Bidae",
                "value": 89.96,
              },
              Object {
                "key": "Hadiya",
                "value": 85.38,
              },
              Object {
                "key": "Abdullah Al Salem",
                "value": 83.43,
              },
              Object {
                "key": "Faiha",
                "value": 83.02,
              },
              Object {
                "key": "Jaber Al Ahmad",
                "value": 78.24,
              },
              Object {
                "key": "Nuzha",
                "value": 74.36,
              },
              Object {
                "key": "Abu Halifa",
                "value": 67.52,
              },
              Object {
                "key": "Sheikh Saad Aviation Terminal",
                "value": 57.89,
              },
              Object {
                "key": "Reggai",
                "value": 47.269999999999996,
              },
              Object {
                "key": "Qortuba",
                "value": 45.3,
              },
              Object {
                "key": "Saad Al Abdullah",
                "value": 36.51,
              },
              Object {
                "key": "Nahda",
                "value": 36.35,
              },
              Object {
                "key": "Mirqab",
                "value": 34.87,
              },
              Object {
                "key": "Fahaheel",
                "value": 34.84,
              },
              Object {
                "key": "Qasser",
                "value": 30.51,
              },
              Object {
                "key": "Doha",
                "value": 20.85,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "axisFormats": Object {
              "x": [Function],
              "y": [Function],
            },
            "data": Array [
              Object {
                "key": "Fridays",
                "value": 83.29,
              },
              Object {
                "key": "Mondays",
                "value": 57.89,
              },
              Object {
                "key": "Saturdays",
                "value": 234.60000000000002,
              },
              Object {
                "key": "Sundays",
                "value": 187.74,
              },
              Object {
                "key": "Thursdays",
                "value": 418.09000000000003,
              },
              Object {
                "key": "Tuesdays",
                "value": 65.38,
              },
              Object {
                "key": "Wednesdays",
                "value": 383.93,
              },
            ],
            "onClick": [Function],
            "x": "key",
            "y": "value",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FlexGridRow
          gutter={48}
          justify="space-around"
        >
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Branch A",
                  "value": 230.59,
                },
                Object {
                  "key": "Branch B",
                  "value": 123.42000000000002,
                },
                Object {
                  "key": "Branch C",
                  "value": 250.42000000000002,
                },
                Object {
                  "key": "Branch D",
                  "value": 230.42000000000002,
                },
                Object {
                  "key": "Branch E",
                  "value": 220.12,
                },
                Object {
                  "key": "Branch F",
                  "value": 375.95,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Rai",
                  "value": 176.21,
                },
                Object {
                  "key": "Qibla",
                  "value": 130.1,
                },
                Object {
                  "key": "Zahra",
                  "value": 112.94,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 91.43,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 89.96,
                },
                Object {
                  "key": "Hadiya",
                  "value": 85.38,
                },
                Object {
                  "key": "Abdullah Al Salem",
                  "value": 83.43,
                },
                Object {
                  "key": "Faiha",
                  "value": 83.02,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 78.24,
                },
                Object {
                  "key": "Nuzha",
                  "value": 74.36,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 67.52,
                },
                Object {
                  "key": "Sheikh Saad Aviation Terminal",
                  "value": 57.89,
                },
                Object {
                  "key": "Reggai",
                  "value": 47.269999999999996,
                },
                Object {
                  "key": "Qortuba",
                  "value": 45.3,
                },
                Object {
                  "key": "Saad Al Abdullah",
                  "value": 36.51,
                },
                Object {
                  "key": "Nahda",
                  "value": 36.35,
                },
                Object {
                  "key": "Mirqab",
                  "value": 34.87,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 34.84,
                },
                Object {
                  "key": "Qasser",
                  "value": 30.51,
                },
                Object {
                  "key": "Doha",
                  "value": 20.85,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
          <BarChart
            axisFormats={
              Object {
                "x": [Function],
                "y": [Function],
              }
            }
            data={
              Array [
                Object {
                  "key": "Fridays",
                  "value": 83.29,
                },
                Object {
                  "key": "Mondays",
                  "value": 57.89,
                },
                Object {
                  "key": "Saturdays",
                  "value": 234.60000000000002,
                },
                Object {
                  "key": "Sundays",
                  "value": 187.74,
                },
                Object {
                  "key": "Thursdays",
                  "value": 418.09000000000003,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 65.38,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 383.93,
                },
              ]
            }
            onClick={[Function]}
            x="key"
            y="value"
          />
        </FlexGridRow>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <BarChart
              axisFormats={
                Object {
                  "x": [Function],
                  "y": [Function],
                }
              }
              data={
                Array [
                  Object {
                    "key": "Branch A",
                    "value": 230.59,
                  },
                  Object {
                    "key": "Branch B",
                    "value": 123.42000000000002,
                  },
                  Object {
                    "key": "Branch C",
                    "value": 250.42000000000002,
                  },
                  Object {
                    "key": "Branch D",
                    "value": 230.42000000000002,
                  },
                  Object {
                    "key": "Branch E",
                    "value": 220.12,
                  },
                  Object {
                    "key": "Branch F",
                    "value": 375.95,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
            <BarChart
              axisFormats={
                Object {
                  "x": [Function],
                  "y": [Function],
                }
              }
              data={
                Array [
                  Object {
                    "key": "Rai",
                    "value": 176.21,
                  },
                  Object {
                    "key": "Qibla",
                    "value": 130.1,
                  },
                  Object {
                    "key": "Zahra",
                    "value": 112.94,
                  },
                  Object {
                    "key": "Al Qusour",
                    "value": 91.43,
                  },
                  Object {
                    "key": "Al Bidae",
                    "value": 89.96,
                  },
                  Object {
                    "key": "Hadiya",
                    "value": 85.38,
                  },
                  Object {
                    "key": "Abdullah Al Salem",
                    "value": 83.43,
                  },
                  Object {
                    "key": "Faiha",
                    "value": 83.02,
                  },
                  Object {
                    "key": "Jaber Al Ahmad",
                    "value": 78.24,
                  },
                  Object {
                    "key": "Nuzha",
                    "value": 74.36,
                  },
                  Object {
                    "key": "Abu Halifa",
                    "value": 67.52,
                  },
                  Object {
                    "key": "Sheikh Saad Aviation Terminal",
                    "value": 57.89,
                  },
                  Object {
                    "key": "Reggai",
                    "value": 47.269999999999996,
                  },
                  Object {
                    "key": "Qortuba",
                    "value": 45.3,
                  },
                  Object {
                    "key": "Saad Al Abdullah",
                    "value": 36.51,
                  },
                  Object {
                    "key": "Nahda",
                    "value": 36.35,
                  },
                  Object {
                    "key": "Mirqab",
                    "value": 34.87,
                  },
                  Object {
                    "key": "Fahaheel",
                    "value": 34.84,
                  },
                  Object {
                    "key": "Qasser",
                    "value": 30.51,
                  },
                  Object {
                    "key": "Doha",
                    "value": 20.85,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
            <BarChart
              axisFormats={
                Object {
                  "x": [Function],
                  "y": [Function],
                }
              }
              data={
                Array [
                  Object {
                    "key": "Fridays",
                    "value": 83.29,
                  },
                  Object {
                    "key": "Mondays",
                    "value": 57.89,
                  },
                  Object {
                    "key": "Saturdays",
                    "value": 234.60000000000002,
                  },
                  Object {
                    "key": "Sundays",
                    "value": 187.74,
                  },
                  Object {
                    "key": "Thursdays",
                    "value": 418.09000000000003,
                  },
                  Object {
                    "key": "Tuesdays",
                    "value": 65.38,
                  },
                  Object {
                    "key": "Wednesdays",
                    "value": 383.93,
                  },
                ]
              }
              onClick={[Function]}
              x="key"
              y="value"
            />,
          ],
          "gutter": 48,
          "justify": "space-around",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "axisFormats": Object {
                "x": [Function],
                "y": [Function],
              },
              "data": Array [
                Object {
                  "key": "Branch A",
                  "value": 230.59,
                },
                Object {
                  "key": "Branch B",
                  "value": 123.42000000000002,
                },
                Object {
                  "key": "Branch C",
                  "value": 250.42000000000002,
                },
                Object {
                  "key": "Branch D",
                  "value": 230.42000000000002,
                },
                Object {
                  "key": "Branch E",
                  "value": 220.12,
                },
                Object {
                  "key": "Branch F",
                  "value": 375.95,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "axisFormats": Object {
                "x": [Function],
                "y": [Function],
              },
              "data": Array [
                Object {
                  "key": "Rai",
                  "value": 176.21,
                },
                Object {
                  "key": "Qibla",
                  "value": 130.1,
                },
                Object {
                  "key": "Zahra",
                  "value": 112.94,
                },
                Object {
                  "key": "Al Qusour",
                  "value": 91.43,
                },
                Object {
                  "key": "Al Bidae",
                  "value": 89.96,
                },
                Object {
                  "key": "Hadiya",
                  "value": 85.38,
                },
                Object {
                  "key": "Abdullah Al Salem",
                  "value": 83.43,
                },
                Object {
                  "key": "Faiha",
                  "value": 83.02,
                },
                Object {
                  "key": "Jaber Al Ahmad",
                  "value": 78.24,
                },
                Object {
                  "key": "Nuzha",
                  "value": 74.36,
                },
                Object {
                  "key": "Abu Halifa",
                  "value": 67.52,
                },
                Object {
                  "key": "Sheikh Saad Aviation Terminal",
                  "value": 57.89,
                },
                Object {
                  "key": "Reggai",
                  "value": 47.269999999999996,
                },
                Object {
                  "key": "Qortuba",
                  "value": 45.3,
                },
                Object {
                  "key": "Saad Al Abdullah",
                  "value": 36.51,
                },
                Object {
                  "key": "Nahda",
                  "value": 36.35,
                },
                Object {
                  "key": "Mirqab",
                  "value": 34.87,
                },
                Object {
                  "key": "Fahaheel",
                  "value": 34.84,
                },
                Object {
                  "key": "Qasser",
                  "value": 30.51,
                },
                Object {
                  "key": "Doha",
                  "value": 20.85,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "axisFormats": Object {
                "x": [Function],
                "y": [Function],
              },
              "data": Array [
                Object {
                  "key": "Fridays",
                  "value": 83.29,
                },
                Object {
                  "key": "Mondays",
                  "value": 57.89,
                },
                Object {
                  "key": "Saturdays",
                  "value": 234.60000000000002,
                },
                Object {
                  "key": "Sundays",
                  "value": 187.74,
                },
                Object {
                  "key": "Thursdays",
                  "value": 418.09000000000003,
                },
                Object {
                  "key": "Tuesdays",
                  "value": 65.38,
                },
                Object {
                  "key": "Wednesdays",
                  "value": 383.93,
                },
              ],
              "onClick": [Function],
              "x": "key",
              "y": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
